dev-tools/deploy/preflight.py

```
@@ -1342,29 +2442,62 @@ def main() -> int:
-            has_required_failure = append_result(check_results, waived_ids, waiver_approver, waiver_ticket, "PREFLIGHT-BACKUP-001", False, "not_applicable", "Promotion attestation missing") or has_required_failure
+            has_required_failure = append_result(check_results, waived_ids, waiver_approver, waiver_ticket, "PREFLIGHT-BACKUP-001", True, "fail", "Recovery compatibility cannot be evaluated because the promotion attestation is missing") or has_required_failure
         else:
-            promotion_status, rollback_mode, promotion_message = promotion_check(Path(promotion_attestation), service_images, root_dir)
-            has_required_failure = append_result(check_results, waived_ids, waiver_approver, waiver_ticket, "PREFLIGHT-PROMOTION-001", True, promotion_status, promotion_message) or has_required_failure
-            if rollback_mode != "roll-forward-only":
-                has_required_failure = append_result(check_results, waived_ids, waiver_approver, waiver_ticket, "PREFLIGHT-BACKUP-001", False, "not_applicable", "Backup readiness is required only for roll-forward-only promotions") or has_required_failure
-            elif not backup_readiness_evidence:
-                has_required_failure = append_result(check_results, waived_ids, waiver_approver, waiver_ticket, "PREFLIGHT-BACKUP-001", True, "fail", "FIREMUD_BACKUP_READINESS_EVIDENCE is required for roll-forward-only promotions") or has_required_failure
-            elif not Path(backup_readiness_evidence).exists():
-                has_required_failure = append_result(check_results, waived_ids, waiver_approver, waiver_ticket, "PREFLIGHT-BACKUP-001", True, "fail", f"Backup-readiness evidence file not found: {backup_readiness_evidence}") or has_required_failure
+            promotion_status, rollback_mode, promotion_message = promotion_check(
```

Ruff (0.15.21)

[warning] Unpacked variable `rollback_mode` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)